src/button/ReactionToggleButton.tsx

```diff
@@ -0,0 +1,295 @@
+          >
+            🖐️
+          </CpdButton>
+        </Tooltip>
+      </section>
+      <div className={styles.verticalSeperator} />
+      <section>
+        {isSearching ? (
+          <>
+            <Form.Root
+              className={styles.searchForm}
+              onSubmit={(e) => e.preventDefault()}
+            >
+              <Search
+                value={searchText}
+                name="reactionSearch"
+                placeholder="Search reactions…"
+                onChange={onSearch}
+                autoFocus
+              />
+              <CpdButton
+                Icon={CloseIcon}
+                size="sm"
+                kind="destructive"
+                onClick={() => setIsSearching(false)}
+              />
+            </Form.Root>
+            <Separator />
+          </>
+        ) : null}
+        <menu>
+          {filteredReactionSet.map((reaction) => (
+            <li className={styles.reactionPopupMenuItem}>
+              <Tooltip label={reaction.name}>
+                <CpdButton
+                  kind="secondary"
+                  className={styles.reactionButton}
+                  key={reaction.name}
+                  disabled={!canReact}
+                  onClick={() => sendRelation(reaction)}
+                >
+                  {reaction.emoji}
+                </CpdButton>
+              </Tooltip>
+            </li>
+          ))}
+          {!isSearching ? (
+            <li key="search" className={styles.reactionPopupMenuItem}>
+              <Tooltip label="Search">
+                <CpdButton
+                  iconOnly
+                  Icon={SearchIcon}
+                  kind="tertiary"
+                  onClick={() => setIsSearching(true)}
+                />
+              </Tooltip>
+            </li>
+          ) : null}
+        </menu>
+      </section>
+    </div>
+  );
+}
+
+interface ReactionToggleButtonProps {
+  rtcSession: MatrixRTCSession;
+  client: MatrixClient;
+}
+
+export function ReactionToggleButton({
+  client,
+  rtcSession,
+}: ReactionToggleButtonProps): ReactNode {
+  const { raisedHands, myReactionId, reactions } = useReactions();
+  const [busy, setBusy] = useState(false);
+  const userId = client.getUserId()!;
+  const isHandRaised = !!raisedHands[userId];
+  const memberships = useMatrixRTCSessionMemberships(rtcSession);
+  const [showReactionsMenu, setShowReactionsMenu] = useState(false);
+
+  const canReact = !reactions[userId];
+
+  const sendRelation = useCallback(
+    async (reaction: ReactionOption) => {
+      const myMembership = memberships.find((m) => m.sender === userId);
+      if (!myMembership?.eventId) {
+        logger.error("Cannot find own membership event");
+        return;
+      }
+      const parentEventId = myMembership.eventId;
+      try {
+        setBusy(true);
+        // @ts-expect-error Trying to send a unspec'd event seems to miss the 3rd overload, need to come back to this.
```

Comment:
This needs https://github.com/matrix-org/matrix-js-sdk/pull/4497 in order to work.